src/Files.App/Actions/FileSystem/CreateAlternateDataStreamAction.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) 2024 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+using Microsoft.UI.Xaml.Controls;
+using System.IO;
+using Windows.Foundation.Metadata;
+
+namespace Files.App.Actions
+{
+	internal sealed class CreateAlternateDataStreamAction : BaseUIAction, IAction
+	{
+		private readonly IContentPageContext context;
+
+		public string Label
+			=> "CreateAlternateDataStream".GetLocalizedResource();
+
+		public string Description
+			=> "CreateAlternateDataStreamDescription".GetLocalizedResource();
+
+		public override bool IsExecutable =>
+			context.HasSelection &&
+			context.CanCreateItem &&
+			(context.ShellPage?.ShellViewModel.WorkingDirectory != Path.GetPathRoot(context.ShellPage?.ShellViewModel.WorkingDirectory)) &&
```

Comment:
Is this condition necessary?

Reply:
It's to account for https://github.com/files-community/Files/issues/16337#issuecomment-2452221219

Reply:
Then this condition should be deleted, because Create ADS will not appear when right-clicking on an empty space even without this condition,
This condition prevents the creation of ADSs for items in the root directory.

Reply:
Fair enough